sweep/sweeper.go

```diff
@@ -21,7 +21,7 @@ type Sweeper struct {
 
 // CreateSweepTx creates an htlc sweep tx.
 func (s *Sweeper) CreateSweepTx(
-	globalCtx context.Context, height int32,
+	globalCtx context.Context, height int32, sequence uint32,
```

Comment:
This function signature isn't ideal. The combination of the `htlc` object and separate data that also partially available in `htlc` is a bit strange. I think that if the pre-image would be set on `htlc` when it becomes known, the function parameters `sequence`, `keyBytes` and `witnessFunc` could be removed from `CreateSweepTx`. Only a (bool) flag indicating whether the timeout or the success path should be taken is required and the rest of the info can be retrieved from `htlc`.

Reply:
Discussed offline, leaving this as is for now.